packages/react-native-renderer/src/__tests__/ReactNativeMount-test.internal.js

```diff
@@ -249,6 +249,106 @@ describe('ReactNative', () => {
     });
   });
 
+  it('should call UIManager.measure on ref.measure', () => {
+    const View = createReactNativeComponentClass('RCTView', () => ({
+      validAttributes: {foo: true},
+      uiViewClassName: 'RCTView',
+    }));
+
+    class Subclass extends ReactNative.NativeComponent {
+      render() {
+        return <View>{this.props.children}</View>;
+      }
+    }
+
+    const CreateClass = createReactClass({
+      mixins: [NativeMethodsMixin],
+      render() {
+        return <View>{this.props.children}</View>;
+      },
+    });
+
+    [View, Subclass, CreateClass].forEach(Component => {
+      UIManager.measure.mockClear();
+
+      let viewRef;
+      ReactNative.render(
+        <Component
+          ref={ref => {
+            viewRef = ref;
+          }}
+        />,
+        11,
+      );
+
+      expect(UIManager.measure).not.toBeCalled();
+
+      const successCallback = jest.fn();
+      viewRef.measure(successCallback);
+
+      expect(UIManager.measure).toHaveBeenCalledTimes(1);
+      expect(UIManager.measure).toHaveBeenCalledWith(
+        expect.any(Number),
+        expect.any(Function),
+      );
```

Comment:
This seems a little janky. Like it's kind of testing internal API details.

Reply:
testing the module’s interactions with the public API of its collaborators, rather than the aggregate behavior of said multiple modules, is how you decouple a test like this from implementation details.

from Xunit patterns book:

> A Mock Object provides the system under test with both indirect input and a way to verify indirect output. All the expectations are configured before the calling of the method under test.

Asserting that no outbound calls to UIManager.measure happen before the viewRef.measure call is a good pinning test and test as documentation IMO.

Reply:
I think my comments were  probably swapped.  _This_ seems janky and the other one seems like it's monkeying with internals a bit too much for my comfort.

Reply:
I'm going to change this to instead have a mock implementation that will throw if passed anything other than a number in the first argument and to call the passed callback with arguments. Then my assertion will just validate that the success callback was called with the expected arguments.